acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/web3j/privacy/PrivacyNet.java

```diff
@@ -0,0 +1,266 @@
+    String orionPublicKeyFileName = String.format("orion_key_%d.pub", i);
+    OrionTestHarness orion;
+    if (otherOrionNodes == null) {
+      // Need conditional because createEnclave will choke if passing in null
+      orion =
+          createEnclave(
+              temporaryFolder, orionPublicKeyFileName, orionPrivateKeyFileName, useKnownOrionKeys);
+    } else {
+      orion =
+          createEnclave(
+              temporaryFolder,
+              orionPublicKeyFileName,
+              orionPrivateKeyFileName,
+              useKnownOrionKeys,
+              otherOrionNodes);
+    }
+
+    PantheonNode pantheon;
+    String keyFilePath;
+    // node 0's file is "key",  every other node includes node number, like "key0"
+    if (i == 0) {
+      keyFilePath = "key";
+    } else {
+      keyFilePath = String.format("key%d", i);
+    }
+    if (isMiningEnabled) {
+      pantheon =
+          pantheonNodeFactory.createPrivateTransactionEnabledMinerNode(
+              nodeName, getPrivacyParameters(orion), keyFilePath);
+    } else {
+      pantheon =
+          pantheonNodeFactory.createPrivateTransactionEnabledNode(
+              nodeName, getPrivacyParameters(orion), keyFilePath);
+    }
+    String orionPubKey =
+        Files.asCharSource(orion.getConfig().publicKeys().get(0).toFile(), UTF_8).read().trim();
+
+    return new PrivacyNode(pantheon, orion, KNOWN_PANTHEON_KEYPAIRS.get(i), orionPubKey);
+  }
+
+  public void startPantheonNodes() {
+    // Todo: Verify  init was called
+    if (nodes == null)
+      throw new IllegalStateException(
+          "Cannot start network nodes.  init method was never called to initialize the nodes");
+    List<PantheonNode> pantheonNodes =
+        nodes.stream().map(sc -> sc.pantheon).collect(Collectors.toList());
+    // cluster.start(pantheonNodes);
+    cluster.start(pantheonNodes.get(0), pantheonNodes.get(1), pantheonNodes.get(2));
+  }
+
+  public void stopOrionNodes() {
+    if (nodes == null) return; // Never started
+    for (PrivacyNode node : nodes) {
+      try {
+        node.orion.getOrion().stop();
+      } catch (RuntimeException e) {
+        LOG.error(
+            String.format(
+                "Error stopping Orion node %s.  Logging and continuing to shutdown other nodes.",
+                node.orion.nodeUrl()),
+            e);
+      }
+    }
+  }
+
+  public void stop() {
+    try {
+      cluster.stop();
+    } catch (RuntimeException e) {
+      LOG.error("Error stopping Pantheon nodes.  Logging and continuing.", e);
+    }
+    try {
+      stopOrionNodes();
+    } catch (RuntimeException e) {
+      LOG.error("Error stopping Orion nodes.  Logging and continuing.", e);
+    }
+  }
+
+  /** Verify that each Orion node has connected to every other Orion */
+  public void verifyAllOrionNetworkConnections() {
+    for (int i = 0; i < nodes.size(); i++) {
+      for (int j = i; j < nodes.size(); j++) {
```

Comment:
int j = i + 1

you do not need to test connectivity between node 0 and node 0...